sfs/sfs.py

```diff
@@ -244,15 +244,15 @@ def read(self, size=-1):
             return self.readall()
         elif size < 0:
             raise ValueError("size must be >= -1")
-        buffer = ctypes.create_string_buffer(size)
-        bytes_read = c_helper.nfs4_pread(self._file, buffer, self._pos, size)
+        buf = ctypes.create_string_buffer(size)
+        bytes_read = c_helper.nfs4_pread(self._file, buf, self._pos, size)
         if bytes_read < 0:
             if c_helper.nfs4_error_num(client) == NFS4ERR_OPENMODE:
                 raise io.UnsupportedOperation("not readable") 
             print("Failed to read file: " + c_helper.nfs4_error_string(client).decode(encoding='utf-8'))
             return
-        self._pos += len(buffer.value)
-        return buffer.value
+        self._pos += bytes_read
+        return bytes(buf[:bytes_read])
```

Comment:
I suspect there might be a more efficient way to to use this buffer, as well as creating buf.
It's necessary to constraint the return values to only `bytes_read` number of bytes.